src/main/java/com/minecolonies/coremod/colony/buildings/AbstractCitizenAssignable.java

```diff
@@ -0,0 +1,253 @@
+    {
+        if (isCitizenAssigned(citizen))
+        {
+            assignedCitizen.remove(citizen);
+            citizen.setWorkBuilding(null);
+            markDirty();
+        }
+    }
+
+    /**
+     * Returns if the {@link CitizenData} is the same as the worker.
+     *
+     * @param citizen {@link CitizenData} you want to compare
+     * @return true if same citizen, otherwise false
+     */
+    public boolean isCitizenAssigned(final CitizenData citizen)
+    {
+        return assignedCitizen.contains(citizen);
+    }
+
+    /**
+     * Returns the first worker in the list.
+     *
+     * @return the EntityCitizen of that worker.
+     */
+    public Optional<EntityCitizen> getMainCitizenEntity()
+    {
+        if (assignedCitizen.isEmpty())
+        {
+            return Optional.empty();
+        }
+        return assignedCitizen.get(0).getCitizenEntity();
+    }
+
+    /**
+     * Returns whether or not the building has a worker.
+     *
+     * @return true if building has worker, otherwise false.
+     */
+    public boolean hasAssignedCitizen()
+    {
+        return !assignedCitizen.isEmpty();
+    }
+
+    /**
+     * Returns the {@link net.minecraft.entity.Entity} of the worker.
+     *
+     * @return {@link net.minecraft.entity.Entity} of the worker
+     */
+    @Nullable
+    public List<Optional<EntityCitizen>> getAssignedEntities()
+    {
+        return assignedCitizen.stream().filter(Objects::nonNull).map(CitizenData::getCitizenEntity).collect(Collectors.toList());
+    }
+
+    /**
+     * Assign the citizen to the current building.
+     *
+     * @param citizen {@link CitizenData} of the worker
+     */
+    public boolean assignCitizen(final CitizenData citizen)
+    {
+        if (assignedCitizen.contains(citizen))
+        {
+            return false;
+        }
+
+        // If we set a worker, inform it of such
+        if (citizen != null)
+        {
+            assignedCitizen.add(citizen);
+        }
+
+        markDirty();
+        return true;
+    }
+
+    /**
+     * Assign the citizen after loading it from NBT.
+     * @param data the citizen data.
+     */
+    public void assignCitizenFromNBtAction(final CitizenData data)
+    {
+        /**
+         * Specific classes will override this.
+         */
+    }
+
+    /**
+     * Returns whether the citizen has this as home or not.
+     *
+     * @param citizen Citizen to check.
+     * @return True if citizen lives here, otherwise false.
+     */
+    public boolean hasAssignedCitizen(final CitizenData citizen)
+    {
+        return assignedCitizen.contains(citizen);
+    }
+
+
+    /**
+     * Checks if the building is full.
+     *
+     * @return true if so.
+     */
+    public boolean isFull()
+    {
+        return assignedCitizen.size() >= getMaxInhabitants();
+    }
+
+    /**
+     * Returns the max amount of inhabitants.
+     *
+     * @return Max inhabitants.
+     */
+    public int getMaxInhabitants()
+    {
+        return 1;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3400)